skimage/restoration/_denoise.py

```diff
@@ -62,8 +66,30 @@ def denoise_bilateral(image, win_size=5, sigma_range=None, sigma_spatial=1,
     >>> astro = astro[220:300, 220:320]
     >>> noisy = astro + 0.6 * astro.std() * np.random.random(astro.shape)
     >>> noisy = np.clip(noisy, 0, 1)
-    >>> denoised = denoise_bilateral(noisy, sigma_range=0.05, sigma_spatial=15)
+    >>> denoised = denoise_bilateral(noisy, sigma_range=0.05,
+    ...                              sigma_spatial=15, multichannel=True)
     """
+    if multichannel:
+        if image.ndim != 3:
+            raise ValueError("Use ``multichannel=False`` for 2D grayscale images. "
+                             "The last axis of the input image must be multiple "
+                             "color channels not another spatial dimension.")
+        elif image.shape[2] not in (3, 4):
+            if image.shape[2] > 4:
```

Comment:
in principle, it's not a problem to have a lot of channels, eg for hyperspectral images. The principle of ``multichannel`` is to transfer to the user the responsibility of whether the last dimension should be considered as channels or not.